### DISCUSSION TITLE
Autocomplete: Show dropdown/search list when MaximumSelectedOptions="1" and an item is already selected

### DISCUSSION THREAD

**Discussion by lasbrojen** [microsoft/fluentui-blazor#2866] [Q&A] [answered]
Hey,

I’m using FluentAutocomplete as a dropdown where the user can select only one item, so I’ve set MaximumSelectedOptions="1".

Currently, when the user wants to change their selection, they have to remove the selected item before the dropdown list/search list appears again.

I’d like the user to be able to open the dropdown and have the option to select a new item without needing to remove the current selection first.

I'm looking for a workaround for that?

❤️

**Comment by vnbaaij** [comment, marked as answer]
That would mean we need to provide code (and test and maintain) that would specifically only run if MaximumSelectedOptions is 1.
I don't think that is a very good practice.

**Reply by lasbrojen** [reply]
Okay 👍

I have a dropdown in my system that functions similarly to the one on your site, allowing users to select a single employee. After making a selection, the user can go back to edit, and ideally, they could simply click on the selected employee to view the list again, even when IsReachedMaxItems is set to true. However, on further reflection, I realize this could add significant complexity. While in my case, I could continue the search using the first letter of the employee’s name, other users of your package might apply different search criteria in OnSearchAsync.

Thank you for the fantastic package and for your time ❤️ 

![image](https://github.com/user-attachments/assets/a29abe9c-179e-4107-94a2-e57e8fdccfc7)

**Comment by lasbrojen** [comment]
Setting: `MaximumSelectedOptions="2"`

And this dirty code:
 ```
private bool _isChanging;
 private void SelectedItemsChanged(IEnumerable<SelectItemQueryModel> items)
 {
     if(_isChanging)
     {
         _isChanging = false;
         return;
     }

     var itemsMaterialized = items.ToList();

     if (itemsMaterialized.Count() > 1)
     {
         _isChanging = true;
         SelectedItems = itemsMaterialized.Where(x => x.Id != SelectedId).ToList();
     } else
         SelectedItems = itemsMaterialized.ToList();
     
     SelectedIdChanged.InvokeAsync(SelectedItems?.FirstOrDefault()?.Id);
 }
```

 Gave me almost what I was looking for:
 
![2024-10-26_00h52_11](https://github.com/user-attachments/assets/e0d449e3-c71a-48cc-972d-d08753cfc325)

**Comment by Trapulo** [comment]
this seems so strange: there isn't any way to have something as (for example) BlazoredTypeahead out of the box?

**Reply by vnbaaij** [reply]
We only have (very) limited resources, so we can't do everything...

Anyways, I think the Combobox comes quite close to what Typeahead offered: https://www.fluentui-blazor.net/Combobox#autocompleteexamples

**Comment by Trapulo** [comment]
I see. Combobox is great for layout and filtering features, but I cannot find a way to search options using an external items source.

Basically what I need is to allow user to select a single object from a list, listing options searching them from a remote api when he types something. This is the way to assign a value to a model property, where the property refers to a master object in the model.
Think to a form to insert an Order, and a field refers to the Customer. I need a control to select a single Customer typing something in the field, and selecting Customer from a list created filtering the full customers list via a call to remote API.
